### PR TITLE
add update-user to user

### PR DESCRIPTION
**Pull request by jnicolas2476**
add update-user to user

### CODE REVIEW

File: backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Controllers/UserController.java [hunk 1]
```diff
@@ -29,5 +33,17 @@ public ResponseEntity<User> saveUser(@RequestBody User user){
...
+   @PutMapping("/{id}")
+    public ResponseEntity<String> updateUser(
+            @PathVariable("id") Long id, 
+            @RequestBody User user) {
+        boolean updated = userService.updateUser(id, user);
+        if (updated) {
+            return ResponseEntity.ok("Usuario actualizado exitosamente");
+        } else {
+            return ResponseEntity.status(404).body("Usuario no encontrado");
+        }
```
Comment (Mosipro1) [inline backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Controllers/UserController.java:46]: in english plis:
`return ResponseEntity.ok("Usuario actualizado exitosamente");
        } else {
            return ResponseEntity.status(404).body("Usuario no encontrado");`

File: backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Service/UserService.java [hunk 2]
```diff
@@ -49,5 +49,16 @@ public User saveUser(User user) {
         user.setPassword(Security.hashPassword(user.getPassword()));
         return userRepository.save(user);
     }
+
+    public boolean updateUser(Long id, User user) {
```
Comment (Mosipro1) [inline backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Service/UserService.java:53]: Integer here, no LONG:
` public boolean updateUser(Integer id, User user) {`

File: backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Service/UserService.java [hunk 3]
```diff
@@ -49,5 +49,16 @@ public User saveUser(User user) {
...
+        if (userRepository.existsById(id)) {
+            User existingUser = userRepository.findById(id).get();
+            existingUser.setPassword(user.getPassword());
```
Comment (alinsan03) [inline backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Service/UserService.java:56]: You have to hash the password, please use the 
`Security.hashPassword(user.getPassword())`

File: backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Controllers/UserController.java [hunk 4]
```diff
@@ -29,5 +33,17 @@ public ResponseEntity<User> saveUser(@RequestBody User user){
...
+
+   @PutMapping("/{id}")
+    public ResponseEntity<String> updateUser(
+            @PathVariable("id") Long id, 
+            @RequestBody User user) {
+        boolean updated = userService.updateUser(id, user);
+        if (updated) {
+            return ResponseEntity.ok("Usuario actualizado exitosamente");
+        } else {
+            return ResponseEntity.status(404).body("Usuario no encontrado");
```
Comment (Mosipro1) [inline backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Controllers/UserController.java:45]: inglishhhhhh

File: backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Controllers/UserController.java [hunk 5]
```diff
@@ -29,5 +33,17 @@ public ResponseEntity<User> saveUser(@RequestBody User user){
     User newUser = userService.saveUser(user);
     return ResponseEntity.status(HttpStatus.CREATED).body(newUser);
    }
+
+   @PutMapping("/{id}")
+    public ResponseEntity<String> updateUser(
+            @PathVariable("id") Long id,
```
Comment (Mosipro1) [inline backend/habragasolina/src/main/java/bo/edu/usfa/gasolina/habragasolina/Controllers/UserController.java:39]: change "Long" with  Integer

**Review by freman226 — COMMENTED** [review]
looks good so far

**Review by freman226 — APPROVED** [review]

**Pull request merged**